components/searchable_input/index.coffee

```diff
@@ -0,0 +1,19 @@
+module.exports = ($el) ->
+  NAMESPACE = 'searchableInput'
+
+  bind: ->
+    $el
+      .on "focus change keyup paste.#{NAMESPACE}", 'input', ({ currentTarget: { value }}) ->
+        $el.attr 'data-state', if value is '' then 'focus' else 'input'
+
+      .on "blur.#{NAMESPACE}", 'input', ->
+        $el.attr 'data-state', 'blur'
+
+      .on "click.#{NAMESPACE}", 'input', (e) ->
+        e.stopPropagation()
```

Comment:
Since I'm detecting events on a pseudo element, which isn't part of the DOM, we have to capture clicks on the parent element. But I still want them on this child input. Hence stopping propagation.